packages/cactus-plugin-ledger-connector-polkadot/src/main/typescript/plugin-ledger-connector-polkadot.ts

```diff
@@ -0,0 +1,901 @@
+  isWeb3SigningCredentialMnemonicString,
+  isWeb3SigningCredentialNone,
+} from "./model-type-guards";
+import {
+  IInvokeContractEndpointOptions,
+  InvokeContractEndpoint,
+} from "./web-services/invoke-contract-endpoint";
+
+export interface IPluginLedgerConnectorPolkadotOptions
+  extends ICactusPluginOptions {
+  logLevel?: LogLevelDesc;
+  pluginRegistry: PluginRegistry;
+  prometheusExporter?: PrometheusExporter;
+  wsProviderUrl: string;
+  instanceId: string;
+  autoConnect?: boolean;
+}
+
+export class PluginLedgerConnectorPolkadot
+  implements
+    IPluginLedgerConnector<
+      DeployContractInkRequest,
+      DeployContractInkResponse,
+      RunTransactionRequest,
+      RunTransactionResponse
+    >,
+    ICactusPlugin,
+    IPluginWebService
+{
+  public static readonly CLASS_NAME = "PluginLedgerConnectorPolkadot";
+  private readonly instanceId: string;
+  private readonly log: Logger;
+  private readonly pluginRegistry: PluginRegistry;
+  public wsProvider: WsProvider | undefined;
+  public api: ApiPromise | undefined;
+  public prometheusExporter: PrometheusExporter;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private autoConnect: false | number | undefined;
+
+  public getOpenApiSpec(): unknown {
+    return OAS;
+  }
+
+  public get className(): string {
+    return PluginLedgerConnectorPolkadot.CLASS_NAME;
+  }
+
+  constructor(public readonly opts: IPluginLedgerConnectorPolkadotOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.pluginRegistry, `${fnTag} options.pluginRegistry`);
+    if (typeof opts.logLevel !== "undefined") {
+      Checks.truthy(opts.logLevel, `${fnTag} options.logLevelDesc`);
+    }
+    Checks.truthy(opts.wsProviderUrl, `${fnTag} options.wsProviderUrl`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    this.pluginRegistry = opts.pluginRegistry;
+    this.prometheusExporter =
+      opts.prometheusExporter ||
+      new PrometheusExporter({ pollingIntervalInMin: 1 });
+    Checks.truthy(
+      this.prometheusExporter,
+      `${fnTag} options.prometheusExporter`,
+    );
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = opts.instanceId;
+    if (opts.autoConnect) {
+      this.autoConnect = 1;
+    }
+    this.setProvider(opts.wsProviderUrl);
+    this.prometheusExporter.startMetricsCollection();
+  }
+
+  public setProvider(wsProviderUrl: string): void {
+    try {
+      this.wsProvider = new WsProvider(wsProviderUrl, this.autoConnect);
+    } catch (e) {
+      throw Error(`Could not create wsProvider. InnerException: + ${e}`);
+    }
+  }
+
+  public async createAPI(): Promise<void> {
+    try {
+      this.api = await ApiPromise.create({ provider: this.wsProvider });
+    } catch (e) {
+      throw Error("Could not create API");
```

Comment:
Please print the error `err` as well (after renaming the var)

Reply:
👍